### ISSUE TITLE
Cursor is blinking when using `nvim_win_set_cursor` in another window

### ISSUE THREAD

**Issue by wojciech-kulik** [neovim/neovim#28023]
### Problem

I'm implementing autoscroll feature for Test Explorer which updates very often. Scrolling this window causes blinking of the cursor:

https://github.com/neovim/neovim/assets/3128467/6b7997cd-c79f-4771-a5c7-ce920eeca183

### Steps to reproduce

```
---Autoscrolls the cursor to the provided row.
---@param row number|nil
local function autoscroll_cursor(row)
  local winnr = M.bufnr and vim.fn.win_findbuf(M.bufnr)[1]

  if winnr and config.cursor_follows_tests and row and last_set_cursor_row ~= row then
    -- V1
    -- vim.api.nvim_win_call(winnr, function()
    --   vim.cmd(tostring(row))
    --   last_set_cursor_row = row
    -- end)

    -- V2
    vim.api.nvim_win_set_cursor(winnr, { row, 0 })
    last_set_cursor_row = row
  end
end
```

I tried both approaches `V1` and `V2` and it happens for both.

### Expected behavior

Cursor should not blink

### Neovim version (nvim -v)

v0.9.5 and v0.10.0

### Vim (not Nvim) behaves the same?

-

### Operating system/version

macOS 14.4

### Terminal name/version

iTerm 2

### $TERM environment variable

xterm-256

### Installation

Homebrew/source code

**Comment by zeertzjq** [comment]
The steps to reproduce don't do anything.

**Comment by wojciech-kulik** [comment]
I updated steps. I tested it with `nvim --noplugin`. I noticed that it's more significant on:
```
NVIM v0.10.0-dev-2675+g2955c921c
Build type: RelWithDebInfo
LuaJIT 2.1.1710088188
```

than

```
NVIM v0.9.5
Build type: Release
LuaJIT 2.1.1710088188
```

**Comment by zeertzjq** [comment]
I think this may be a terminal problem. I tried several terminals. I cannot reproduce this on kitty, XTerm, Alacritty or Konsole, but can reproduce this on GNOME Terminal or Terminator.

**Comment by luukvbaal** [comment]
Yeah I also tried 4 different terminals but don't see any blinking. If it's more significant on master that is likely due to #27858.

**Comment by wojciech-kulik** [comment]
I tired it now on Kitty and Alacritty and it doesn't happen. I was able to reproduce it on iTerm2.

**Comment by wojciech-kulik** [comment]
I guess the label "needs:repro" can be removed?

**Comment by luukvbaal** [comment]
@wojciech-kulik now that #28023 is merged, can you confirm that #27858 indeed only made it worse and was not the sole reason it happened in the first place? I.e. is this still an issue or can it be closed?
